### PR TITLE
feat: Multimodal block display + Steamship agent

### PR DESCRIPTION
**Pull request by eob**
This PR adds two separate (but related) features:

1. Basic support in the UI for multimodal blocks (think Notion-style) as the agent response type. String completions, as provided before, are rendered as a text block.
2. A Steamship companion of Rick (from Rick & Morty), which supports audio and image responses in addition to text.

Note: We've wire this agent to a hosted copy of Rick w/ custom voice & character knowledge, for which you'll need an API key to access -- we can DM that key for your use.

image

### CODE REVIEW

**Comment by ykhli** [comment]
Thanks so much for this @eob! I tried it out on codespaces but for some reason the audio / image generation didn't work. I can check out the branch locally to see why thats the case. 

Would be super helpful if you could add more instructions to creating a user's own agent on steamship / have a ready-to-use agent people can deploy easily when they first launch this app too!

**Comment by EniasCailliau** [comment]
@ykhli I've added a PR that adds more instructions for creating a user's own agent on Steamship. Users will be able to either use the pre-deployed instance featured in this PR or can instantiate their own companions with the instructions in the readme. 

Here's the PR #57

**Comment by EniasCailliau** [comment]
@ykhli audio/image generation is now fixed. Sorry for the confusion! 

image

File: README.md [hunk 1]
```diff
@@ -113,15 +113,24 @@ e. **Upstash API key**
...
 
+g. **Steamship API key**
+
+You can connect Steamship agent instance as if it were an LLM with personality, vector storage, and tools built-in. To do so:
+
+- Sign in to [Steamship](https://steamship.com/account)
+- Copy the API key from your account settings page
+- Add it as the `STEAMSHIP_API_KEY` variable
+
+Next, [build an agent](https://www.steamship.com/learn/agent-guidebook), create your own instance of it, and connect it in `companions.json` using the *Rick* example as a guide.
```
Comment (jenniferli23) [inline README.md:132]:
```suggestion
g. **Steamship API key**

You can connect a Steamship agent instance as an LLM with personality, voice and image generation capabilities built in. It also includes its own vector storage and tools. To do so:

- Create an account on [Steamship](https://steamship.com/account)
- Copy the API key from your account settings page
- Add it as the `STEAMSHIP_API_KEY` variable 

If you'd like to create your own character personality, add a custom voice, or use a different image model, visit [Steamship Agent Guidebook](https://www.steamship.com/learn/agent-guidebook), create your own instance and connect it in `companions.json` using the *Rick* example as a guide.
```

File: companions/companions.json [hunk 2]
```diff
@@ -4,27 +4,40 @@
     "title": "I love talking about books and games",
     "imageUrl": "/alex.png",
     "llm": "chatgpt",
-    "phone": "OPTIONAL_COMPANION_PHONE_NUMBER"
+    "phone": "OPTIONAL_COMPANION_PHONE_NUMBER",
+    "telegram": "OPTIONAL_TELEGRAM_LINK"
```
Comment (jenniferli23) [inline companions/companions.json:8]:
```suggestion
 
```

File: companions/companions.json [hunk 3]
```diff
@@ -4,27 +4,40 @@
...
   },
   {
     "name": "Rosie",
     "title": "I'm a house robot who became aware",
     "imageUrl": "/rosie.png",
     "llm": "vicuna13b",
-    "phone": "OPTIONAL_COMPANION_PHONE_NUMBER"
+    "phone": "OPTIONAL_COMPANION_PHONE_NUMBER",
+    "telegram": "OPTIONAL_TELEGRAM_LINK"
```
Comment (jenniferli23) [inline companions/companions.json:16]: Please remove the rest of the telegram option except for Rick

File: companions/companions.json [hunk 4]
```diff
@@ -4,27 +4,40 @@
...
+  },
+  {
+    "name": "Rick",
+    "title": "I can generate voice and pictures.",
```
Comment (jenniferli23) [inline companions/companions.json:20]:
```suggestion
    "title": "I can generate voice and pictures. You also can talk to me on Telegram",
```

**Review by jenniferli23 — CHANGES_REQUESTED** [review]
Looks great! Just a few suggestions for easy following.

**Comment by eob** [comment]
Thanks for the feedback! Merged in your suggestions and pulled the `telegram` link on all of them for now. We can send another PR that adds it back in, conditionally on the field being present, later if there's interest.

**Comment by eob** [comment]
@jenniferli23 one last update -- Enias did some prompt engineering to get the image quality much better, and added in the telegram link for just the Rick bot.

I think this is ready to merge, but lmk if you need anymore changes!

image

**Pull request merged**